app/services/parsers/text.rb

```diff
@@ -9,7 +9,7 @@ def initialize(document)
 
     # Enumerable chunk records
     def chunks
-      chonker.chunk(text)
+      chonker.chunk(self)
```

Comment:
This seems odd, but anything else I can think of would introduce too much complexity to be worth it.

Once we support a couple more filetypes, the chunker-parser classes may need a bit of a refactor.

Reply:
Yes.  Even having 2 different classes may not be useful/necessary.